wip-trampoline.md

```diff
@@ -0,0 +1,780 @@
+# Trampoline Onion Routing
+
+This document describes the additions we make to the existing onion routing to enable the use of trampoline nodes.
+
+As the network grows, more bandwidth and storage will be required to keep an up-to-date view of the whole network.
+Finding a payment path will also require more computing power, making it unsustainable for constrained devices.
+
+Constrained devices should only keep a view of a small part of the network and leverage trampoline nodes to route
+payments. This proposal still uses an onion created by the payer so it doesn't sacrifice privacy (in most cases it
+will even provide a bigger anonymity set).
+
+Nodes that keep track of the whole network should advertise support for `trampoline_routing`. This is an opportunity
```

Comment:
I was pretty sure you would comment something like that, especially after reading your last email on routing ;)

I totally agree that in the long run, we don't want/need to ask trampoline nodes to sync the whole network and accurately estimate fees (we have retries to make up for inaccuracy and failures). But we must take this problem one step at a time, and I think this goes in the right direction. This proposal doesn't force trampoline nodes to sync the whole network and I think it slowly goes towards a packet-switched, internet-like network where trampoline nodes would act as AS routers.

I'm not afraid of the cryptographic/mac part: we will find a way to make it work. So bottom line is: I agree with you on the long-term goals and I think this is a first step towards that.

Reply:
Okay.  I am afraid I am skimming the details for now.  It seems fine for a first cut.

Reply:
I don't know, I would still rephrase it to "nodes that want to act as trampoline nodes" or something like that. Since it is not a requirement for trampoline nodes to keep track of the whole network, "Nodes that keep track of the whole network" seems unnecessarily confusing.

Reply:
Good suggestion. I'll do another big pass on this PR once I'm done with the implementation in eclair, I learnt a lot while implementing so there are a few points that have changed slightly.